src/main/java/org/prebid/server/auction/CurrencyConversionService.java

```diff
@@ -0,0 +1,198 @@
+package org.prebid.server.auction;
+
+import io.vertx.core.Vertx;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.http.HttpClient;
+import io.vertx.core.http.HttpClientResponse;
+import io.vertx.core.json.Json;
+import org.apache.commons.collections4.MapUtils;
+import org.prebid.server.currency.proto.response.CurrencyConversionRates;
+import org.prebid.server.exception.PreBidException;
+import org.prebid.server.util.HttpUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Service for price currency conversion between currencies.
+ */
+public class CurrencyConversionService {
+
+    private static final Logger logger = LoggerFactory.getLogger(CurrencyConversionService.class);
+
+    private static final String DEFAULT_BID_CURRENCY = "USD";
+    private final String currencyServerUrl;
+    private final HttpClient httpClient;
+
+    private Map<String, Map<String, BigDecimal>> latestCurrencyRates;
+
+    public CurrencyConversionService(String currencyServerUrl, long period, HttpClient httpClient, Vertx vertx) {
+        this.currencyServerUrl = HttpUtil.validateUrl(Objects.requireNonNull(currencyServerUrl));
+        this.httpClient = Objects.requireNonNull(httpClient);
+        Objects.requireNonNull(vertx).setPeriodic(validatePeriod(period), aLong -> populatesLatestCurrencyRates());
+        populatesLatestCurrencyRates();
+    }
+
+    /**
+     * Validates consumed period value.
+     */
+    private long validatePeriod(long period) {
+        if (period < 1) {
+            throw new IllegalArgumentException("Period for updating rates must be positive value");
+        }
+        return period;
+    }
+
+    /**
+     * Updates latest currency rates by making a call to currency server.
+     */
+    private void populatesLatestCurrencyRates() {
+        httpClient.getAbs(currencyServerUrl, this::handleResponse);
+    }
+
+    private void handleResponse(HttpClientResponse response) {
+        final int statusCode = response.statusCode();
+        if (statusCode != 200) {
+            logger.warn("Currency server response code is {0}", statusCode);
+            return;
+        }
+        response.bodyHandler(this::handleBody)
+                .exceptionHandler(CurrencyConversionService::handleException);
+    }
+
+    /**
+     * Parses body content and populates latest currency rates.
+     */
+    private  void handleBody(Buffer buffer) {
+        try {
+            final Map<String, Map<String, BigDecimal>> receivedCurrencyRates =
+                    Json.mapper.readValue(buffer.toString(), CurrencyConversionRates.class).getConversions();
+            if (receivedCurrencyRates == null) {
```

Comment:
Not sure why we need to throw exception in this case. It would be way more straightforward to do just:
```
    if (recievedCurrencyRates != null) {
        latestCurrencyRates = receivedCurrencyRates;
    }
```

Reply:
I thought it might be helpful to have a warning logged when response json contains null _conversions_ attribute.